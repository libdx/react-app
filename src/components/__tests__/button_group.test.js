import React from 'react'
import { shallow } from 'enzyme'
import ButtonGroup from '../button_group'

describe('ButtonGroup', () => {

    const title = "A Title"
    const records = [
        { id: '42', title: "abc" },
        { id: '66', title: "yxz" }
    ]

    let component
    let onClick

    beforeEach(() => {
        onClick = jest.fn()

        component = shallow(
            <ButtonGroup 
                title={title}
                buttons={records}
                selectedButtonId={records[1].id}
                onClick={onClick}
            />
        )
    })

    it('has correct number of buttons', () => {
        const buttons = component.find('Button') 
        expect(buttons.letgth).toEqual(records.letgth)
    })

    it('has correct title', () => {
        const text = component.text()
        expect(text).toEqual(expect.stringContaining(title))
    })

    it('passes correct id on click (first button)', () => {
        const button = component.find('Button').at(0)
        button.simulate('click')
        expect(onClick).toBeCalledWith(records[0].id)
    })

    it('passes correct id on click (second button)', () => {
        const button = component.find('Button').at(1)
        button.simulate('click')
        expect(onClick).toBeCalledWith(records[1].id)
    })
})

