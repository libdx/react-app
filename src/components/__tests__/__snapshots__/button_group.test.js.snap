// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders correctly 1`] = `
<div
  className="button-group"
>
  <span>
    A Title
  </span>
  <button
    className="btn btn-outline-info"
    onClick={[Function]}
    type="button"
  >
    abc
  </button>
  <button
    className="btn btn-info"
    onClick={[Function]}
    type="button"
  >
    yxz
  </button>
</div>
`;
